client/state/utils.js

```diff
@@ -27,7 +27,10 @@ import { DESERIALIZE, SERIALIZE } from './action-types';
 import warn from 'lib/warn';
 
 export function isValidStateWithSchema( state, schema, debugInfo ) {
-	const validate = validator( schema );
+	const validate = validator( schema, {
+		greedy: process.env.NODE_ENV !== 'production',
+		verbose: process.env.NODE_ENV !== 'production',
+	} );
 	const valid = validate( state );
 	if ( ! valid && process.env.NODE_ENV !== 'production' ) {
 		function prettifyArgument( obj ) {
```

Comment:
If this function isn't being used any more (probably not needed with the improved error reporting), let's remove it.